library/kani_macros/src/lib.rs

```diff
@@ -138,9 +161,11 @@ mod sysroot {
             #[kanitool::proof]
         );
 
-        assert!(attr.is_empty(), "#[kani::proof] does not take any arguments currently");
-
         if sig.asyncness.is_none() {
+            assert!(
```

Comment:
Can you please move this check to the parser? Also, please make it a user friendly error.

Reply:
The options parser does not know what function `#[proof]` is applied to, so I cannot move it there, but made it more user-friendly as requested.